vowpalwabbit/core/include/vw/core/reductions/eigen_memory_tree.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+#include "vw/core/example.h"
+#include "vw/core/rand_state.h"
+
+#include <list>
+#include <unordered_map>
+#include <vector>
+
+namespace VW
+{
+namespace reductions
+{
+VW::LEARNER::base_learner* eigen_memory_tree_setup(VW::setup_base_i& stack_builder);
+
+namespace eigen_memory_tree
+{
+using emt_feats = std::vector<std::pair<int, float>>;
```

Comment:
Are these hashes? If so, then we should use the same type as elsewhere `uint64_t`